### ISSUE TITLE
Avoid importing generateUtilityClass from @mui/material to reduce module weight

### ISSUE THREAD

**Issue by rbayliss** [mui/mui-x#6394]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

When I import date picker from `@mui/x-date-pickers/DatePicker`, my bundle contains all of `@mui/material`, due to the import of `generateUtilityClass` and `generateUtilityClasses` from `@mui/material`. I'm not actually sure how to resolve that, but I see that there have been some preliminary stabs at it (#3094). 

It would be great for my use case if I didn't have the full weight of @mui/material when I use a date picker.

### Motivation 🔦

I'm building a NextJS 12 application to be deployed on AWS Lambda. In doing that, I'm using `next build` with the `standalone` target.  This does some magic to trace what code is actually used, and only include that in the final build. So I've got a finite size of the built application due to Lambda limitations, so it would be great not to carry around this extra weight. In practice, we haven't hit our limit yet, but we're close. I've tried a bunch of things to do improve tree-shaking, but always end up back at the limitation that this module brings in all of `@mui/material`. 

So in summary, removing that import of `@mui/material` would help me sleep better knowing I'm not about to lose the ability to deploy code :).

**Comment by LukasTy** [comment]
@rbayliss I don't have much experience with next, maybe you could test a [PR build](https://pkg.csb.dev/mui/mui-x/commit/d8a59aca/@mui/x-date-pickers) with `@mui/material` imports replaced by `@mui/utils`.
I've tested those changes on a `vite` project with `rollup-plugin-visualizer` -> zero effect on tree-shaken packages.
Do not forget that:
- DatePicker will need quite a few components from `@mui/material`—the minimal amount of components is shown in the screenshot below
- If you are using `DatePicker` with `TextField` from mui—this components has its own fair share of transitive components

![Screenshot 2022-10-10 at 11 23 20](https://user-images.githubusercontent.com/4941090/194825075-adb0c593-0e14-4996-903a-b85fd5ed206e.png)
